packages/compass-aggregations/src/components/focus-mode/focus-mode-modal-header.tsx

```diff
@@ -322,13 +340,16 @@ export default connect(
       pipelineBuilder: {
         stageEditor: { stages },
       },
+      searchIndexes: { isSearchIndexesSupported },
     } = state;
     const stage = stages[stageIndex] as StoreStage;
 
     return {
       stageIndex,
       isEnabled: !stage?.disabled,
-      insight: stage ? getInsightForStage(stage, env) : undefined,
+      stage,
```

Comment:
Do we need to pass the stage here, if we pass also all stages and stageIndex? Or it can be accessed directly in the FocusModeModalHeader component?

Reply:
The stages we pass to the component are mapped with only index and operator properties. The stage passed here is full stage.